cfme/fixtures/candu.py

```diff
@@ -90,3 +123,367 @@ def candu_db_restore(temp_appliance_extended_db):
+    # Verify that metrics are present in the metrics or metric_rollups tables.
+    interval_name = 'realtime' if table_name == 'metrics' else 'hourly'
+    vm_name = provider.data['cap_and_util']['chargeback_vm']
+    client = appliance.db.client
+    ems = client['ext_management_systems']
+    metric_table = client[table_name]
+
+    with client.transaction:
+        metric_count = (
+            client.session.query(metric_table.id)
+            .join(ems, metric_table.parent_ems_id == ems.id)
+            .filter(metric_table.capture_interval_name == interval_name,
+                metric_table.resource_name == vm_name,
+                ems.name == provider.name,
+                metric_table.timestamp >= date.today())
+            .count()
+        )
+
+    return metric_count >= expected_count
+
+# Retrieve resource usage values from metric_rollups table.
+# Chargeback and metering reports are done on hourly and daily rollups, not realtime values.
+# We capture C&U data and force hourly rollups by running these commands through
+# the Rails console.
+#
+# Metering reports differ from chargeback reports in that metering reports:
+#   1.) report only resource usage, not costs
+#   2.) report the sum total of resource usage instead of average usage.
+@pytest.fixture(scope="module")
+def resource_usage(vm_ownership, appliance, provider):
+    cpu_used = 0
+    memory_used = 0
+    network_used = 0
+    disk_used = 0
+    storage_used = 0
+    consumed_hours = 0
+    cpu_alloc = 0
+    memory_alloc = 0
+    storage_alloc = 0
+
+    vm_name = provider.data['cap_and_util']['chargeback_vm']
+    metrics = appliance.db.client['metrics']
+    rollups = appliance.db.client['metric_rollups']
+    perf_states = appliance.db.client['vim_performance_states']
+    ems = appliance.db.client['ext_management_systems']
+
+    logger.info('Deleting METRICS DATA from metrics and metric_rollups tables')
+    with appliance.db.client.transaction:
+        appliance.db.client.session.query(metrics).delete()
+        appliance.db.client.session.query(rollups).delete()
+        appliance.db.client.session.query(perf_states).delete()
+
+    result = appliance.ssh_client.run_rails_command(
+        f"\"vm = Vm.where(:ems_id => {provider.id}).where(:name => {repr(vm_name)})[0];"
+        "start_time = 2.hour.ago.utc.change(:sec => 0);"
+        "end_time = Time.now.utc.change(:sec => 0);"
+        "vm.perf_capture('realtime', start_time, end_time)\"")
+    assert result.success, f"Failed to capture VM C&U data: {result.output}"
+
+    wait_for(verify_metrics, [appliance, provider, 'metrics', 170], timeout=600,
+        fail_condition=False, message="Waiting for VM realtime data")
+
+    # Disable C&U collection after fetching data, so that the server doesn't automatically collect
+    # additional C&U data and mess with our reports.
+    appliance.server.settings.disable_server_roles(
+        'ems_metrics_coordinator', 'ems_metrics_collector')
```

Comment:
consider putting these roles into a tuple to keep the disable/enable calls consistent.